### PR TITLE
[IT-937] Setup bucket notification

### PR DESCRIPTION
**Pull request by zaro0508**
Configure bucket notification to trigger lambda function.

### CODE REVIEW

**Comment by clarisse-lau** [comment]
If the bucket is deployed in a different account from the lambda, permission must be given for it to invoke the lambda (i.e. step 6 in the [setup](https://github.com/Sage-Bionetworks/s3-synapse-sync/blob/master/README.md)) before the notification can be configured. Will that present an issue here?

**Comment by zaro0508** [comment]
> If the bucket is deployed in a different account from the lambda, permission must be given for it to invoke the lambda (i.e. step 6 in the [setup](https://github.com/Sage-Bionetworks/s3-synapse-sync/blob/master/README.md)) before the notification can be configured. Will that present an issue here?

This is the template to create the bucket.  Any bucket that is created from this template will lambda notifications setup (step 6).  When you want to create new buckets (in any account) just use this template to create the bucket and everything is ready to go.

File: templates/htan-synapse-bucket.yaml [hunk 1]
```diff
@@ -64,6 +64,12 @@ Resources:
             AllowedOrigins: ['*']
             AllowedMethods: [GET, POST, PUT, HEAD]
             MaxAge: 3000
+      NotificationConfiguration:
+        LambdaConfigurations:
+          - Event: "s3:ObjectCreated:*"
+            Function: !Ref S3SynapseSyncFunctionRoleArn
```
Comment (clarisse-lau) [inline templates/htan-synapse-bucket.yaml:70]: We need the ARN for the lambda function itself, rather than its role

Comment (zaro0508) [reply templates/htan-synapse-bucket.yaml]: good catch. fixed

**Pull request merged**